src/docfx/build/context/Context.cs

```diff
@@ -61,17 +64,36 @@ public Context(string outputPath, ErrorLog errorLog, Docset docset, Docset fallb
             DependencyResolver = new DependencyResolver(
                 docset,
                 fallbackDocset,
+                dependencyDocsets.
+                    ToDictionary(
+                        k => k.Key,
+                        v => v.Value.docset,
+                        PathUtility.PathComparer),
                 Input,
                 BuildScope,
                 BuildQueue,
                 GitCommitProvider,
                 BookmarkValidator,
-                restoreGitMap,
                 DependencyMapBuilder,
                 _xrefResolver,
                 TemplateEngine);
         }
 
+        private static Dictionary<string, (Docset docset, bool inScope)> LoadDependencies(Docset docset, RestoreGitMap restoreGitMap)
```

Comment:
consider not putting logic inside `Context`, `BuildScope` and `DependencyResolver` can build their own private data structure.

Reply:
they are sharing data structure, so they don;t need to build their own. what's the concern here? It's just a local variable

Reply:
I moved it out of context, treat it like `docset`, `fallbackdocset`.